src/test/java/spoon/test/trycatch/TryCatchTest.java

```diff
@@ -442,4 +444,52 @@ public void testTryWithVariableAsResource() {
 		tryStmt.removeResource(ctResource);
 
 	}
+
+	@Nested
+	class AddCatcherAt {
+		@Test
+		void addsCatcherAtTheSpecifiedPosition() {
+			// contract: the catcher should be added at the specified position
+			// arrange
+			Factory factory = createFactory();
+
+			CtTry tryStatement = factory.createTry();
+
+			CtCatch first = createCatch(factory, IOException.class);
+			CtCatch second = createCatch(factory, ExceptionInInitializerError.class);
+			CtCatch third = createCatch(factory, NoSuchMethodException.class);
+
+			// act
+			tryStatement
+				.addCatcherAt(0, third)
+				.addCatcherAt(0, first)
+				.addCatcherAt(1, second);
+
+			// assert
+			assertThat(tryStatement.getCatchers(), contains(first, second, third));
+		}
+
+		@Test
+		void throwsOutOfBoundsException_whenPositionIsOutOfBounds() {
+			// contract: `addCatcherAt` should throw an out-of-bounds exception when the specified position is out of
+			// bounds of the catcher collection
+
+			// arrange
+			Factory factory = createFactory();
+
+			CtTry tryStatement = factory.createTry();
+			CtCatch catcherAtWrongPosition = createCatch(factory, Exception.class);
+
+			// act & assert
+			assertThrows(IndexOutOfBoundsException.class,
+					() -> tryStatement.addCatcherAt(1, catcherAtWrongPosition));
+		}
+
+		private <T extends Throwable> CtCatch createCatch(Factory factory, Class<T> typeToCatch) {
```

Comment:
I made it type stricter by replacing `T` with `<T extends Throwable>`

Reply:
:+1: